src/study/system-model/include/antares/study/system-model/library.h

```diff
@@ -63,7 +67,7 @@ class Library
     std::string description_;
 
     std::unordered_map<std::string, PortType> portTypes_;
-    std::unordered_map<std::string, Model> models_;
+    std::shared_ptr<std::unordered_map<std::string, Model>> models_;
```

Comment:
Using directly an `std::unordered_map<std::string, Model>` leads to the following compile error
```
/home/omnesflo/Antares_Simulator/src/solver/modelConverter/modelConverter.cpp: In function ‘std::vector<Antares::Study::SystemModel::Port> Antares::Solver::ModelConverter::convertPorts(const Antares::Solver::ModelParser::Model&)’:
/home/omnesflo/Antares_Simulator/src/solver/modelConverter/modelConverter.cpp:145:46: warning: unused parameter ‘model’ [-Wunused-parameter]
  145 |   const Antares::Solver::ModelParser::Model& model)
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
In file included from /usr/include/c++/11/ext/alloc_traits.h:34,
                 from /usr/include/c++/11/bits/basic_string.h:40,
                 from /usr/include/c++/11/string:55,
                 from /usr/include/c++/11/stdexcept:39,
                 from /home/omnesflo/Antares_Simulator/src/solver/modelConverter/modelConverter.cpp:24:
/usr/include/c++/11/bits/alloc_traits.h: In instantiation of ‘static constexpr void std::allocator_traits<std::allocator<_CharT> >::construct(std::allocator_traits<std::allocator<_CharT> >::allocator_type&, _Up*, _Args&& ...) [with _Up = std::pair<const std::__cxx11::basic_string<char>, Antares::Study::SystemModel::Model>; _Args = {const std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, Antares::Study::SystemModel::Model>&}; _Tp = std::__detail::_Hash_node<std::pair<const std::__cxx11::basic_string<char>, Antares::Study::SystemModel::Model>, true>; std::allocator_traits<std::allocator<_CharT> >::allocator_type = std::allocator<std::__detail::_Hash_node<std::pair<const std::__cxx11::basic_string<char>, Antares::Study::SystemModel::Model>, true> >]’:
/usr/include/c++/11/bits/hashtable_policy.h:1878:36:   required from ‘std::__detail::_Hashtable_alloc<_NodeAlloc>::__node_type* std::__detail::_Hashtable_alloc<_NodeAlloc>::_M_allocate_node(_Args&& ...) [with _Args = {const std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, Antares::Study::SystemModel::Model>&}; _NodeAlloc = std::allocator<std::__detail::_Hash_node<std::pair<const std::__cxx11::basic_string<char>, Antares::Study::SystemModel::Model>, true> >; std::__detail::_Hashtable_alloc<_NodeAlloc>::__node_ptr = std::allocator<std::__detail::_Hash_node<std::pair<const std::__cxx11::basic_string<char>, Antares::Study::SystemModel::Model>, true> >::value_type*]’
/usr/include/c++/11/bits/hashtable_policy.h:168:32:   required from ‘std::__detail::_AllocNode<_NodeAlloc>::__node_type* std::__detail::_AllocNode<_NodeAlloc>::operator()(_Arg&&) const [with _Arg = const std::pair<const std::__cxx11::basic_string<char>, Antares::Study::SystemModel::Model>&; _NodeAlloc = std::allocator<std::__detail::_Hash_node<std::pair<const std::__cxx11::basic_string<char>, Antares::Study::SystemModel::Model>, true> >; std::__detail::_AllocNode<_NodeAlloc>::__node_type = std::__detail::_AllocNode<std::allocator<std::__detail::_Hash_node<std::pair<const std::__cxx11::basic_string<char>, Antares::Study::SystemModel::Model>, true> > >::__node_type]’
/usr/include/c++/11/bits/hashtable.h:1306:20:   required from ‘void std::_Hashtable<_Key, _Value, _Alloc, _ExtractKey, _Equal, _Hash, _RangeHash, _Unused, _RehashPolicy, _Traits>::_M_assign(_Ht&&, const _NodeGenerator&) [with _Ht = const std::_Hashtable<std::__cxx11::basic_string<char>, std::pair<const std::__cxx11::basic_string<char>, Antares::Study::SystemModel::Model>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, Antares::Study::SystemModel::Model> >, std::__detail::_Select1st, std::equal_to<std::__cxx11::basic_string<char> >, std::hash<std::__cxx11::basic_string<char> >, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<true, false, true> >&; _NodeGenerator = std::__detail::_AllocNode<std::allocator<std::__detail::_Hash_node<std::pair<const std::__cxx11::basic_string<char>, Antares::Study::SystemModel::Model>, true> > >; _Key = std::__cxx11::basic_string<char>; _Value = std::pair<const std::__cxx11::basic_string<char>, Antares::Study::SystemModel::Model>; _Alloc = std::allocator<std::pair<const std::__cxx11::basic_string<char>, Antares::Study::SystemModel::Model> >; _ExtractKey = std::__detail::_Select1st; _Equal = std::equal_to<std::__cxx11::basic_string<char> >; _Hash = std::hash<std::__cxx11::basic_string<char> >; _RangeHash = std::__detail::_Mod_range_hashing; _Unused = std::__detail::_Default_ranged_hash; _RehashPolicy = std::__detail::_Prime_rehash_policy; _Traits = std::__detail::_Hashtable_traits<true, false, true>]’
/usr/include/c++/11/bits/hashtable.h:1421:16:   required from ‘std::_Hashtable<_Key, _Value, _Alloc, _ExtractKey, _Equal, _Hash, _RangeHash, _Unused, _RehashPolicy, _Traits>::_Hashtable(const std::_Hashtable<_Key, _Value, _Alloc, _ExtractKey, _Equal, _Hash, _RangeHash, _Unused, _RehashPolicy, _Traits>&) [with _Key = std::__cxx11::basic_string<char>; _Value = std::pair<const std::__cxx11::basic_string<char>, Antares::Study::SystemModel::Model>; _Alloc = std::allocator<std::pair<const std::__cxx11::basic_string<char>, Antares::Study::SystemModel::Model> >; _ExtractKey = std::__detail::_Select1st; _Equal = std::equal_to<std::__cxx11::basic_string<char> >; _Hash = std::hash<std::__cxx11::basic_string<char> >; _RangeHash = std::__detail::_Mod_range_hashing; _Unused = std::__detail::_Default_ranged_hash; _RehashPolicy = std::__detail::_Prime_rehash_policy; _Traits = std::__detail::_Hashtable_traits<true, false, true>]’
/usr/include/c++/11/bits/unordered_map.h:181:7:   required from here
/usr/include/c++/11/bits/alloc_traits.h:518:28: error: no matching function for call to ‘construct_at(std::pair<const std::__cxx11::basic_string<char>, Antares::Study::SystemModel::Model>*&, const std::pair<const std::__cxx11::basic_string<char>, Antares::Study::SystemModel::Model>&)’
  518 |           std::construct_at(__p, std::forward<_Args>(__args)...);
      |           ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
```
So far, using an `std::shared_ptr` is the only solution I found. This is probably related to move/copy issues on some classes, unfortunately there's no way to trace the issue to the relevant classes.

Reply:
if I'm right and error occurs at `modelConverter.cpp:24`, why not remove the corresponding line : 
```C++
#include <stdexcept>
```
which seems to be useless ?

Reply:
done